tests/minifier.js

```diff
@@ -840,6 +840,54 @@
     equal(minify(input, { removeOptionalTags: true }), output);
     equal(minify(input, { removeOptionalTags: true, removeEmptyElements: true }), output);
 
+    input = ' <html></html>';
+    output = ' ';
+    equal(minify(input, { removeOptionalTags: true }), output);
+    output = '';
+    equal(minify(input, { collapseWhitespace: true, removeOptionalTags: true }), output);
+
+    input = '<html> </html>';
+    output = ' ';
+    equal(minify(input, { removeOptionalTags: true }), output);
+    output = '';
+    equal(minify(input, { collapseWhitespace: true, removeOptionalTags: true }), output);
+
+    input = '<html></html> ';
+    output = ' ';
+    equal(minify(input, { removeOptionalTags: true }), output);
+    output = '';
+    equal(minify(input, { collapseWhitespace: true, removeOptionalTags: true }), output);
+
+    input = ' <html><body></body></html>';
+    output = ' ';
+    equal(minify(input, { removeOptionalTags: true }), output);
+    output = '';
+    equal(minify(input, { collapseWhitespace: true, removeOptionalTags: true }), output);
+
+    input = '<html> <body></body></html>';
+    output = ' ';
+    equal(minify(input, { removeOptionalTags: true }), output);
+    output = '';
+    equal(minify(input, { collapseWhitespace: true, removeOptionalTags: true }), output);
+
+    input = '<html><body> </body></html>';
+    output = '<body> ';
+    equal(minify(input, { removeOptionalTags: true }), output);
+    output = '';
+    equal(minify(input, { collapseWhitespace: true, removeOptionalTags: true }), output);
+
+    input = '<html><body></body> </html>';
+    output = ' ';
+    equal(minify(input, { removeOptionalTags: true }), output);
+    output = '';
+    equal(minify(input, { collapseWhitespace: true, removeOptionalTags: true }), output);
+
+    input = '<html><body></body></html> ';
+    output = ' ';
```

Comment:
Hm, it's probably more theoretical than anything, but I wonder if we really care about leaving space in cases like this...